.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+name: Linting and Unit Tests
+
+# only run this workflow on new commits to main
+# or PRs into main
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  schedule:
+    # Run every Monday morning at 11:00a UTC, 6:00a CST
+    - cron: '0 11 * * 1'
+
+jobs:
+  test:
+    name: ${{ matrix.task }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - task: linting
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Set up Python 3.7
+        if: matrix.task != 'todo-checks'
+        uses: s-weigand/setup-conda@v1.0.3
+        with:
+          python-version: 3.7
+      - name: linting
+        if: matrix.task == 'linting'
+        shell: bash
+        run: |
+          pip install --upgrade black flake8 mypy
```

Comment:
Shall we pin this to specific versions? That way, we would not have issues happening for CI and not happening in dev env.

Reply:
I think that for things like linters, which don't impact our application code, it's better to let them float so you keep getting new features. I don't think that the pain of managing environments and pinning things is worth it for CI-only things like linters.

Reply:
for example, about two weeks ago `pylint` added a check for this PEP on "method chaining" (https://www.python.org/dev/peps/pep-3134/#explicit-exception-chaining) that makes the tracebacks from errors more accurate.

If we had pinned `pylint` in the projects that are using it (like `prefect-saturn`), we never would have known about that and the library would be a tiny tiny bit worse as a result

Reply:
IMO, it's *not* CI only; we use it during dev, and CI is make sure if one forget to run linter/formater before merge, then if CI runs on whatever random version I'm not aware of, it will be PITA to review log output to only then know it was version mismatch. I understand that running that on a schedule basis helps to know about it before happens, but doesn't solve, specially if new tooling releases happen between scheduled runs.

We have many projects running in separated repos, and then we can't use same versioning for tooling in all of them since they are totally separated and meant to be released independently.

Reply:
I agree with James that the latest linters are the best to use. It seems like newer versions of linters should be more strict about what they expect, so if you get a failure from a linter on CI and look at the logs and see that there is some new error that you weren't seeing locally, I don't think that is very painful.

Reply:
@wreis are you convinced or do you want to bring it to the group in coffee chat tomorrow? I don't want to merge this is you still have objections

Reply:
@jameslamb thank you!

Using latest linters is indeed best, thus we achieve having higher quality with better libs as result as you said James. I was just thinking if we could solve the issue for having to deal with version mismatch along the path, no strong objection.

Reply:
ok sounds good, thanks! I've found that these linters are very cheap to install (not a lot of other dependencies, not a large library, nothing OS-specific), so so far it hasn't been too bad to just keep up with the latest versions. If they start releasing more frequently or take on some expensive dependencies, we should revisit that.